### PR TITLE
Add a nccl-env modifier to manage NCCL environment variables

### PR DESCRIPTION
**Pull request by douglasjacobsen**

### CODE REVIEW

File: var/ramble/repos/builtin/modifiers/nccl-env/modifier.py [hunk 1]
```diff
@@ -0,0 +1,714 @@
+# Copyright 2022-2024 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from ramble.modkit import *  # noqa: F403
```
Comment (linsword13) [inline var/ramble/repos/builtin/modifiers/nccl-env/modifier.py]: nit: the noqa comment should not be needed.

File: var/ramble/repos/builtin/modifiers/nccl-env/modifier.py [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+        "nccl_nsocks_perthread",
+        default="",
+        description="Specification of number of sockets opened by each helper thread.",
+        modes=["standard"],
+    )
+
+    modifier_variable(
+        "nccl_cross_nic",
+        default="",
+        values=[0, 1, 2],
```
Comment (linsword13) [inline var/ramble/repos/builtin/modifiers/nccl-env/modifier.py:71]: Does this essentially make it a required variable?

Comment (douglasjacobsen) [reply var/ramble/repos/builtin/modifiers/nccl-env/modifier.py:71]: No, this just allows `ramble info --type modifiers nccl-env` to print "suggested values" for this variable. It's more for documentation than any kind of enforcement.

File: var/ramble/repos/builtin/modifiers/nccl-env/modifier.py (hunk 1 above)
Comment (douglasjacobsen) [reply var/ramble/repos/builtin/modifiers/nccl-env/modifier.py]: Thanks, fixed.

**Review by linsword13 — APPROVED** [review]

**Pull request merged**